src/conf/datatypes.h

```diff
@@ -21,6 +21,17 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+typedef enum {
+    NONE = 0,
+    BMS_COMM_TIMEOUT = 1,
+    BMS_OVER_TEMP = 2,
+    CELL_OVER_VOLTAGE = 3,
+    CELL_UNDER_VOLTAGE = 4,
+    CELL_OVER_TEMP = 5,
+    CELL_UNDER_TEMP = 6,
+    CELL_BALANCE = 7
+} BMS_FAULT_CODES;
```

Comment:
This enum doesn't belong here, this header is for conf data types only. I understand this is very little code, but I think just add a bms module (bms.h and bms.c) anyway.

Also, please use `CamelCase` for data type names and add a common prefix to all the enum items, as they are in the global namespace.

Reply:
Put in bms.h and bms.c and used camel case.